grove/alpha/simon/simon.py

```diff
@@ -14,440 +14,316 @@
-             row added that maintains the property described above.
-    :rtype: 2darray
-    """
-    n = len(z)
-    while np.any(z != 0):  # while z is not all zeros
-        if len(W) == 0:
-            W = z
-            W = W.reshape(1, n)
-            break
-        msb_z = most_significant_bit(z)
-
-        # Search for a row to insert z into,
-        # so that it has an earlier significant bit than the row below
-        # and a later one than the row above (when reading left-to-right)
-        got_to_end = True
-        for row_num in range(len(W)):
-            row = W[row_num]
-            msb_row = most_significant_bit(row)
-            # if the row as the same msb as z,
-            # set z to the bitwise xor of z and the current row
-            # as it will be guaranteed to still be orthogonal to s
-            if msb_row == msb_z:
-                z = np.array([z[i] ^ row[i] for i in range(n)])
-                got_to_end = False
-                break
-            # if the row has a greater msb than z,
-            # then this is the row to z insert above
-            elif msb_row > msb_z:
-                W = np.insert(W, row_num, z, 0)
-                got_to_end = False
-                break
-        # if z has a greater msb than all rows,
-        # insert it to the bottom of the array
-        if got_to_end:
-            W = np.insert(W, len(W), z, 0)
-
-    return W
-
-
-def make_square_row_echelon(W):
-    """
-    Make :math:`\\mathbf{\\mathit{W}}`
-    into a square matrix for Simon's algorithm, satisfying a few criteria.
-
-    :param 2darray W: an :math:`(n-1)\\times n` array of 0s and 1s in
-                      row-echelon form such that all rows are orthogonal
-                      to some length :math:`n` vector
-                      of 0s and 1s :math:`\\mathbf{s}`.
-    :return: a two-element tuple. The first element is
-             an :math:`n\\times n` square array identical
-             to :math:`\\mathbf{\\mathit{W}}` except with one row added.
-             That row is chosen to keep :math:`\\mathbf{\\mathit{W}}`
-             in row-echelon form. The second element is the row that
-             the new row is in, where the top row is at index 0.
-    :rtype: tuple
+class Simon(object):
     """
-    n = len(W) + 1
-
-    # Generate one final vector that is not orthonormal to the mask s
-    # can do by adding a vector with a single 1
-    # that can be inserted so that diag(W) is all ones
-    insert_row_num = 0
-    while insert_row_num < n - 1 and W[insert_row_num][insert_row_num] == 1:
-        insert_row_num += 1
-
-    new_row = np.zeros(shape=(n,), dtype=int)
-    new_row[insert_row_num] = 1
-    W = np.insert(W, insert_row_num, new_row, 0)
-
-    return W, insert_row_num
+    `Simon's algorithm`_ is amongst the earliest algorithms providing an exponential speedup of a
+    computation using a Quantum Computer vs. a Classical Computer.
```

Comment:
I don't think `quantum computer/classical computer` needs to be capitalized. Also, it might be good to note it's an exponential speedup in a particular computational model. Namely it's relative to an oracle.

Reply:
done ... removed all of that and referenced the docs